### DISCUSSION TITLE
Check access on adding new source

### DISCUSSION THREAD

**Discussion by umerkle** [NuGet/Home#12959] [Ideas]
I think 'nuget' and 'dotnet nuget' do not check wether the supplied credentials for a private nuget source are ok when adding a new source.

If you later try to access the source, there maybe will be a 401.

Wouldn't it be good to test the credentials and verify access to the source when adding it?

**Comment by zivkan** [comment]
https://github.com/NuGet/Home/issues/new?assignees=&labels=Type%3AFeature%2CTriage%3AUntriaged&projects=&template=FEATURE.yml

Checking credentials seems like a good idea, and I think that any `dotnet nuget add source` (even without credentials) would be interesting to make a sample request, just to make sure the source works (no typo in URL). However, if these commands start returning non-zero exit codes (or writing to stderr), then I'm worried it will break a lot of CI scripts. But if this proposal was made before the existing commands ever existed, I'd support the idea 100%. A feature proposal could discuss risks and mitigations for a potential breaking change

**Reply by umerkle** [reply]
Added a new issue as proposed by you:
[Issue](https://github.com/NuGet/Home/issues/12963)
